src/helpers/response-status.helper.ts

```diff
@@ -0,0 +1,36 @@
+import { HttpStatus } from '@nestjs/common';
+import { isAxiosError } from 'axios';
+
+/**
+ * Checks if the error object is an AxiosError and has a response status matching the specified status.
+ * @param {Object} options - An object containing the error and status.
+ * @param {Error} options.error - The error object to check.
+ * @param {HttpStatus} options.status - The status to compare against.
+ * @returns {boolean} - Returns true if the error is an AxiosError and has a matching response status, otherwise false.
+ */
+export const statusCheck = ({ error, status }: { error: Error; status: HttpStatus }): boolean => isAxiosError(error) && error.response?.status === status;
```

Comment:
We know that this error is `AxiosError` on entering here (Although, I do agree with how it's written here and think we should check for axios error in each of these functions,... If that's the case, I made a mistake with this code and the error should be `unknown` until we check for axios error -- as technically that's the type when handling errors)

Reply:
The error being sent to these functions is of type `Error` under type `KnownError`.

Reply:
Odd -- Okay happy to leave this as is